logfire/_internal/integrations/aiohttp_client.py

```diff
@@ -2,10 +2,18 @@
 
 from opentelemetry.instrumentation.aiohttp_client import AioHttpClientInstrumentor
 
+from logfire import Logfire
 
-def instrument_aiohttp_client(**kwargs: Any):
+
+def instrument_aiohttp_client(logfire_instance: Logfire, **kwargs: Any):
     """Instrument the `aiohttp` module so that spans are automatically created for each client request.
 
     See the `Logfire.instrument_aiohttp_client` method for details.
     """
-    AioHttpClientInstrumentor().instrument(**kwargs)  # type: ignore[reportUnknownMemberType]
+    AioHttpClientInstrumentor().instrument(  # type: ignore[reportUnknownMemberType]
+        **{
+            'tracer_provider': logfire_instance.config.get_tracer_provider(),
```

Comment:
I'm torn on how to do this. On the one hand I don't want to completely prevent users from passing custom providers. We even do that in our own code (passing a NoopMeterProvider to disable fastapi metrics), and I've recommended another user to do the same. On the other hand I don't want to encourage or document it.

Reply:
I mean we're using kwargs already so this seems like a step better